### PR TITLE
refactor: move domain check into small helper util

### PR DESCRIPTION
**Pull request by steveiliop56**
## Summary by CodeRabbit

* **Bug Fixes**
  * Strengthened OAuth redirect URI validation with strict hostname equivalence, optional scheme/port enforcement, safer subdomain handling, and improved support for internationalized hostnames.
  * Updated static access-control domain matching to rely on validated domain equivalence and more robust subdomain/name fallback behavior.
* **New Features**
  * Added a reusable domain validation utility that normalizes hostnames (including IDNA) and can optionally enforce scheme and port rules.
* **Tests**
  * Added comprehensive test coverage for hostname normalization and domain/URL validation edge cases.
* **Chores**
  * Ensured container builds include shared `pkg` code in both dev and distribution images.

### CODE REVIEW

File: pkg/validators/domain_validator.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+func (v *DomainValidator) getEffectivePort(u *url.URL) string {
+	if u.Port() != "" {
+		return u.Port()
+	}
+	if u.Scheme == "https" {
+		return "443"
+	}
+	return "80"
```
Comment (Rycochet) [inline pkg/validators/domain_validator.go]: This is assuming that only http/s is coming in - the code is written pretty agnostically, so wouldn't it be safer to actively check for `http` to give port `80`, and throw an error for unknown scheme otherwise?

edit: Maybe still return `80`, but also log it - not sure on the best choice...

**Review by Rycochet — DISMISSED** [review]

**Comment by steveiliop56** [comment]
@coderabbitai review

**Review by Rycochet — APPROVED** [review]
